src/service/invoice-service.ts

```diff
@@ -397,16 +397,26 @@ export default class InvoiceService {
       });
     }
 
-    if (update.description) base.description = update.description;
-    if (update.addressee) base.addressee = update.addressee;
-    if (update.street) base.addressee = update.street;
-    if (update.postalCode) base.postalCode = update.postalCode;
-    if (update.city) base.postalCode = update.city;
-    if (update.country) base.postalCode = update.country;
-    if (update.reference) base.reference = update.reference;
-    if (update.attention) base.attention = update.attention;
-    if (update.date) base.date = new Date(update.date);
+    const { description, addressee, street, postalCode, city, country, reference, attention, date } = update;
+
+    const updated = Invoice.create(
+      Object.fromEntries(
+        Object.entries({
+          description,
+          addressee,
+          street,
+          postalCode,
+          city,
+          country,
+          reference,
+          attention,
+          date: date ? new Date(date) : undefined,
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        }).filter(([_, value]) => value !== undefined),
+      ),
+    );
```

Comment:
Why not just do `await Invoice.update({ id: base.id }, update)`? You only still need to parse the date to an actual date, but that method looks much cleaner and easier to understand to me.

Reply:
You would still need to filter undefinea

Reply:
Heb een nieuwe suggestie gedaan voor je

Reply:
But is `UpdateInvoiceParams` not a "subset" of `Invoice`? Then you can just use `Invoice.update` method, as undefined objects should be ignored? Only if they are `null`, they should be set to `null` in the database. For any attributes that still need to be parsed (like date), you can use the spread operator, like

```
const { date, ...props } = update;
await Invoice.update({ id: base.id }, { ...props, date: date ? new Date(date) : undefined });
```

Reply:
that works, thanks!